src/main/java/com/sinthoras/visualprospecting/gui/xaeromap/rendersteps/OreVeinRenderStep.java

```diff
@@ -0,0 +1,119 @@
+
+public class OreVeinRenderStep implements InteractableRenderStep {
+
+	private final OreVeinLocation oreVeinLocation;
+	private final ResourceLocation depletedTextureLocation = new ResourceLocation(Tags.MODID, "textures/depleted.png");
+	private final IIcon blockStoneIcon = Blocks.stone.getIcon(0, 0);
+	private final double iconSize = 32;
+	private double iconX;
+	private double iconY;
+
+	public OreVeinRenderStep(OreVeinLocation veinPosition) {
+		oreVeinLocation = veinPosition;
+	}
+
+	@Override
+	public void draw(GuiScreen gui, double cameraX, double cameraZ, double scale) {
+		final double iconSizeHalf = iconSize / 2;
+		final double scaleForGui = Math.max(1.0D, scale);
+		this.iconX = (oreVeinLocation.getBlockX() - 0.5 - cameraX) * scaleForGui - iconSizeHalf;
+		this.iconY = (oreVeinLocation.getBlockZ() - 0.5 - cameraZ) * scaleForGui - iconSizeHalf;
+
+		GL11.glPushMatrix();
+		GL11.glTranslated(oreVeinLocation.getBlockX() - 0.5 - cameraX, oreVeinLocation.getBlockZ() - 0.5 - cameraZ, 0);
+		GL11.glScaled(1 / scaleForGui, 1 / scaleForGui, 1);
+		DrawUtils.drawQuad(blockStoneIcon, -iconSizeHalf, -iconSizeHalf, iconSize, iconSize, 0xFFFFFF, 255);
+
+		DrawUtils.drawQuad(oreVeinLocation.getIconFromPrimaryOre(), -iconSizeHalf, -iconSizeHalf, iconSize, iconSize, oreVeinLocation.getColor(), 255);
+
+		if(!oreVeinLocation.drawSearchHighlight() || oreVeinLocation.isDepleted()) {
+			DrawUtils.drawGradientRect(-iconSizeHalf, -iconSizeHalf, iconSizeHalf, iconSizeHalf, 0, 0x96000000, 0x96000000);
+			if(oreVeinLocation.isDepleted()) {
+				DrawUtils.drawQuad(depletedTextureLocation, -iconSizeHalf, -iconSizeHalf, iconSize, iconSize, 0xFFFFFF, 255);
+			}
+		}
+
+		if(scale >= Utils.journeyMapScaleToLinear(Config.minZoomLevelForOreLabel) && !oreVeinLocation.isDepleted()) {
+			final int fontColor = oreVeinLocation.drawSearchHighlight() ? 0xFFFFFFFF : 0xFF7F7F7F;
+			String text = I18n.format(oreVeinLocation.getName());
+			DrawUtils.drawSimpleLabel(gui, text, 0, -iconSizeHalf - gui.mc.fontRenderer.FONT_HEIGHT - 5, fontColor, 0xB4000000, true);
+		}
+
+		if(oreVeinLocation.isActiveAsWaypoint()) {
+            final double thickness = iconSize / 8;
+            final int color = 0xFFFFD700;
+            DrawUtils.drawGradientRect(-iconSizeHalf - thickness, -iconSizeHalf - thickness, iconSizeHalf, -iconSizeHalf, 0, color, color);
+            DrawUtils.drawGradientRect(iconSizeHalf, -iconSizeHalf - thickness, iconSizeHalf + thickness, iconSizeHalf, 0, color, color);
+            DrawUtils.drawGradientRect(-iconSizeHalf, iconSizeHalf, iconSizeHalf + thickness, iconSizeHalf + thickness, 0, color, color);
+            DrawUtils.drawGradientRect(-iconSizeHalf - thickness, -iconSizeHalf, -iconSizeHalf, iconSizeHalf + thickness, 0, color, color);
+        }
+
+		GL11.glPopMatrix();
+	}
+
+	@Override
+	public boolean isMouseOver(double mouseX, double mouseY, double cameraX, double cameraZ, double scale) {
+		final double scaleForGui = Math.max(1.0D, scale);
```

Comment:
Just drop the `D`

Reply:
Best to transform the mouseX, mouseY in the InteractiveLayerRenderer once

Reply:
Subtracting the camera x/y there makes sense, pretty sure the rest of the magic transformations have to stay since they are different for basically every call